emily/cdk/lib/emily-stack.ts

```diff
@@ -318,6 +321,61 @@ export class EmilyStack extends cdk.Stack {
             sourceArn: api.arnForExecuteApi(),
         });
 
+        // Only add the custom domain name it it's specified.
+        let customRootDomainNameRoot = EmilyStackUtils.getCustomRootDomainName();
+        let hostedZoneId = EmilyStackUtils.getHostedZoneId();
+        if (customRootDomainNameRoot !== undefined) {
+            // Error if the hosted zone ID is not provided.
+            if (hostedZoneId === undefined) {
+                throw new Error("Custom domain name specified but hosted zone ID not provided.");
+            }
+
+            // Make the custom domain name. Add the stage name extension ot the domain name
+            // if it's not what we consider the "production" stage.
+            const customDomainName = EmilyStackUtils.getStageName() === Constants.PROD_STAGE_NAME
+                ? customRootDomainNameRoot
+                : `${EmilyStackUtils.getStageName()}.${customRootDomainNameRoot}`;
+
+            // Get zone.
+            const hostedZone = route53.HostedZone.fromHostedZoneAttributes(this, "HostedZone", {
+                hostedZoneId: hostedZoneId,
+                zoneName: customDomainName,
+            });
+
+            // Get certificate.
+            const certificate = new certificatemanager.Certificate(this, "DomainCertificate", {
+                domainName: customDomainName,
+                validation: certificatemanager.CertificateValidation.fromDns(hostedZone),
+            });
+
+            // Create a custom domain.
+            const customDomain = new apig.DomainName(this, "CustomDomain", {
+                domainName: customDomainName,
+                certificate: certificate,
+                // If the endpoint is in us-east-1 then we'll use EDGE because it's a better faster
+                // option globally. If the stack is in any region other than us-east-1 you'll need
+                // to make the certificate in us-east-1 independently and then reference it here if you
```

Comment:
The complexity of creating resources across regions (the certificate in this case) is just not worth it, but for prod we're going to deploy in `us-east-1`.

If you're saying it makes a big difference I can set `beta` to deploy in `us-east-1` so that it's deployed with an `EDGE` endpoint type.

Reply:
No no I was just agreeing with you about the fact that if another regions is needed probably the region type makes sense and the effort yes it is a bit an overkill